src/radio/radio-group.component.ts

```diff
@@ -51,10 +52,40 @@ import { RadioChange } from "./radio-change.class";
 			[attr.aria-labelledby]="ariaLabelledby"
 			[ngClass]="{
 				'cds--radio-button-group--vertical': orientation === 'vertical',
-				'cds--radio-button-group--label-left': labelPlacement === 'left'
+				'cds--radio-button-group--label-left': labelPlacement === 'left',
+				'cds--radio-button-group--invalid' : !warn && invalid,
+				'cds-radio-button-group--warning': !invalid && warn
 			}">
 			<ng-content></ng-content>
 		</div>
+		<div class="cds--radio-button__validation-msg">
+			<div *ngIf="!warn && invalid">
+				<svg
+					cdsIcon="warning--filled"
+					size="16"
+					class="cds--radio-button__invalid-icon"
+					>
+				</svg>
+				<ng-container *ngIf="!isTemplate(invalidText)" class="cds--form-requirement">{{ invalidText }}</ng-container>
+				<ng-template *ngIf="isTemplate(invalidText)" [ngTemplateOutlet]="invalidText"></ng-template>
+			</div>
+			<div *ngIf="!invalid && warn">
+				<svg
+					cdsIcon="warning--alt--filled"
+					class="cds--radio-button__invalid-icon cds--radio-button__invalid-icon--warning"
+					size="16">
+				</svg>
+				<ng-container *ngIf="!isTemplate(warnText)" class="cds--form-requirement">{{warnText}}</ng-container>
+				<ng-template *ngIf="isTemplate(warnText)" [ngTemplateOutlet]="warnText"></ng-template>
+			</div>
+		</div>
+		<div
+			*ngIf="helperText && !invalid && !warn"
+			class="cds--form__helper-text"
+			[ngClass]="{'cds--form__helper-text--disabled': disabled}">
+			<ng-container *ngIf="!isTemplate(helperText)">{{helperText}}</ng-container>
+			<ng-template *ngIf="isTemplate(helperText)" [ngTemplateOutlet]="helperText"></ng-template>
+		</div>
```

Comment:
Can we update the template to match React where we wrap the radio buttons in a fieldset? That's a more appropriate way to group the radio buttons.